workspaces/local-cli/src/shared/coverage.ts

```diff
@@ -0,0 +1,389 @@
+      colors.grey.underline('Expected'),
+      colors.grey.underline('Percent Coverage'),
+    ],
+  });
+
+  function colorProperly(full: boolean, partial: boolean, none: boolean) {
+    return (string: string): string => {
+      if (full) {
+        return string;
+      } else if (none) {
+        return makeRed(string);
+      } else if (partial) {
+        return makeYellow(string);
+      }
+      return string;
+    };
+  }
+
+  const bodyCoverageColorer = colorProperly(
+    totalN === totalD,
+    totalN < totalD,
+    totalN === 0
+  );
+
+  const endpointCoverageColorer = colorProperly(
+    totalEndpointN === totalEndpointD,
+    totalEndpointN < totalEndpointD,
+    totalEndpointN === 0
+  );
+
+  resultTable.push(
+    [
+      'Documented Body Coverage',
+      totalN.toString(),
+      totalD.toString(),
+      Math.round((totalN / (totalD || 1)) * 100).toString() + '%',
+    ].map(bodyCoverageColorer),
+    [
+      'Documented Endpoint Coverage',
+      totalEndpointN.toString(),
+      totalEndpointD.toString(),
+      Math.round((totalEndpointN / (totalEndpointD || 1)) * 100).toString() +
+        '%',
+    ].map(endpointCoverageColorer)
+  );
+
+  console.log(resultTable.toString());
+}
+
+interface KnownEndpoint {
+  method: string;
+  pathId: string;
+  fullPath: string;
+  descriptor: SpecEndpointDescriptor;
+}
+
+async function getSpecEndpoints(paths: IPathMapping): Promise<KnownEndpoint[]> {
+  const events = await fs.readJSON(paths.specStorePath);
+  const { eventStore, rfcState, rfcService, rfcId } = universeFromEvents(
+    events
+  );
+
+  const queries = Queries(eventStore, rfcService, rfcId);
+  const cachedQueryResults = getCachedQueryResults(queries);
+  const shapesResolvers = queries.shapesResolvers();
+  const endpoints = queries.endpoints();
+  const allEndpoints = endpoints.map((i: any) => ({
+    ...i,
+    fullPath: queries.absolutePath(i.pathId),
+    descriptor: createEndpointDescriptor(i, cachedQueryResults),
+  }));
+
+  return sortBy(allEndpoints, (i) => i.fullPath);
+}
+
+async function getReport(paths: IPathMapping, captureId: string): Promise<any> {
+  const capturesDirectory = path.join(paths.capturesPath, captureId);
+
+  const entries = await fs.readdir(capturesDirectory);
+
+  const coverageFiles = entries
+    .filter((x) => x.startsWith('coverage-'))
+    .map((x) => path.join(capturesDirectory, x));
+
+  const reportsToMerge = await Promise.all(
+    coverageFiles.map(async (i) => {
+      const coverageJSON = await fs.readJSON(i);
+      const asScala = opticEngine.CoverageReportJsonDeserializer.fromJs(
+        coverageJSON
+      );
+      return asScala;
+    })
+  );
+
+  const report = opticEngine.CoverageReportBuilder.emptyReport();
+  //merge in all the reports
+  reportsToMerge.forEach((i) => report.merge(i));
+
+  // const finalReport = opticEngine.CoverageReportJsonSerializer.toJs(report);
+
+  return report;
+}
+
+export function getCachedQueryResults(queries: any) {
```

Comment:
I imagine this would be in @useoptic/domain-utilities if it is common across cli and web ui

Reply:
I don't feel entirely comfortable factoring it out because it is slightly different than the other use case in the UI. It's meant to be used by this feature only for the time being so I'll remove the 'export' to make that clear.